### ISSUE TITLE
[Binary pixel masks from XML annotations] 

### ISSUE THREAD

**Issue by Josuerinho** [paip-2019/challenge#56]
Hi all!

I was following the guidelines described [here](https://paip2019.grand-challenge.org/Dataset/) about how to properly generate the masks for the `test set` so they can be compared with the masks of the `training set`. I believe I understand all the points but the last one:

> - Filling in empty holes and removing small fractions of tissue smaller than 10 pixels in size (regarding them as noises) on the Tissue masks

So the filling empty wholes part, what does exactly mean? Because if there is a hole or fracture in the tissue, for example, I believe it should come back as background or noise when the tissue is masked with the RGB(235, 210, 235) filter and then applied to the mask generated from the XML. So wouldn't in this case the whole be already 'filled up' with class 0 (or background) after this step?

Thanks for your help!!

**Comment by PAIP-challenge** [comment]
Sorry for the late reply.

There will be some empty holes in the tissue masks even after the thresholding by RGB(235, 210, 235) value.
If we let these holes be background(non-tissue), there will be some noisy parts in the tissue mask.
Therefore, we made these holes to be a part of the tissue mask as these holes are located in the mass of tissue.

Thanks,

**Comment by Josuerinho** [comment]
Hi! Thanks for your reply. That is not what it's shown in the tissue masks though. Whenever you open a mask of a WSI that is part of the training folder, you can see that wherever there is a fracture or whole in between the tissue, that also appears in the corresponding masks (and it's coded as 0 or background). And that makes sense to me. However, what I don't understand, is how was that done. When you apply the xml to mask, all the masks generated don't have wholes inside the tissue areas. All the regions are completely filled up. So the validation masks have less "resolution" or granularity when compared to the training masks. How did you manage to add also this information to the training masks?
Thanks!

**Comment by PAIP-challenge** [comment]
Hello, I hope I understood your question well!
First, I would like to suggest you apply the same threshold value used for training data processing for validation data too.
And then, follow the same processing rules for annotation mask generation.
So, there will be tissue masks that have small holes as in training tissue masks.
Using these tissue masks, I think you can generate labels for validation data too!

If you have further questions, please contact us at paip.challenge@gmail.com.
Thanks,
